### PR TITLE
private/marco/slideshow for review 1

### PR DESCRIPTION
**Pull request by mcecchetti**
1.  **slideshow: use LayerRenderer gl context for rendering shape effects**
2.  **slideshow: provide partial shape effects support for 2d context**
3.  **slideshow: remove duplicated code for creating transition**
4.  **slideshow: SlideShowPresenter: removed unused code**
5.  **slideshow: provide support for interactive animation sequence**
6. **slideshow: server side watermarking**
7. **slideshow: provide support for color animations**
8. **slideshow: restore text fields caching**
9. **slideshow: update animated layer info on slide start**
10. **slideshow: text field: link placeholder/content through text shape hash**

note for reviewers: 
- better to review one commit at a time (and read commit messages)
- commit 5 needs a core patch: https://gerrit.libreoffice.org/c/core/+/175821
- commit 7 needs core patch: https://gerrit.libreoffice.org/c/core/+/176169
- commits 8,9,10 need core patches: 
   - https://gerrit.libreoffice.org/c/core/+/175568
   - https://gerrit.libreoffice.org/c/core/+/175569
   - https://gerrit.libreoffice.org/c/core/+/175570
   - https://gerrit.libreoffice.org/c/core/+/176166
   - https://gerrit.libreoffice.org/c/core/+/176167
   - https://gerrit.libreoffice.org/c/core/+/176168

### CODE REVIEW

File: browser/src/slideshow/engine/AnimatedElement.ts [hunk 1]
```diff
@@ -619,11 +596,9 @@ class AnimatedElement {
 		]);
 		renderingContext.setTransform(transform);
 
-		const aElement = applied
-			? this.tfCanvas.transferToImageBitmap()
-			: this.aBaseElement;
+		const aElement = this.aBaseElement;
 		console.debug(
-			`AnimatedElement(${this.sId}).updateLayer:
+			`AnimatedElement(${this.sId}).renerLayer2d:
```
Comment (quikee) [inline browser/src/slideshow/engine/AnimatedElement.ts:601]: I guess this should've been renderLayer2d

Comment (mcecchetti) [reply browser/src/slideshow/engine/AnimatedElement.ts:601]: yep, thanks :-)

**Comment by mcecchetti** [comment]
**added a new commit**: 

> slideshow: provide support for interactive animation sequence

**Comment by mcecchetti** [comment]
Added a new commit: [slideshow: server side watermarking](https://github.com/CollaboraOnline/online/pull/10349/commits/8fcc591062e4fe27ab706824ea089fff721bf791)

**Review by quikee — APPROVED** [review]
I went through and looks good to me. Core patches are also all merged.

**Pull request merged**